### PR TITLE
Fix nodejs application issue with mongodb

### PR DESCRIPTION
**Pull request by jcpowermac**
- Since the template is being process and applied with each
  job run the secret is also being updated since it contains generate
  values from the template.  The mongodb DC is not again rolled out
  and the secret is not updated.  To resolve this the secret is
  delete from the processed template.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -48,17 +48,35 @@ pipeline {
...
+                                    "TAG=${env.TAG}",
+                                    "IMAGESTREAM_TAG=${params.IMAGE_STREAM_LATEST_TAG}",
+                                    "REGISTRY_PROJECT=${params.REGISTRY_PROJECT}",
+                                    "REGISTRY=${params.REGISTRY_URI}")
+
+                            /* If the project's secret exists it must be removed at least in
+                             * the case of using a database along with an application.
+                             * Issue: Since the template is being process and applied with each
+                             * job run the secret is also being updated since it contains generate
+                             * values from the template.
```
Comment (codificat) [inline Jenkinsfile:68]: This makes me wonder if we should take this one step further: do we need to process/apply the whole template on every run? The update to the other objects in the template might not be as harmful as the secret's, but isn't it unnecessary anyway?

**Review by codificat — COMMENTED** [review]
The change looks good to me as a fix for this particular issue with the secret in the sample application/template, but wondering if we shouldn't take a different approach instead and reconsider the whole template processing on each pipeline run as a more generic solution?

**Comment by jcpowermac** [comment]
@codificat ok take a look at the changes when you get a chance.

**Comment by jcpowermac** [comment]
Found a bug during testing with Jenkins 2.73.2...back to WIP.

**Review by codificat — APPROVED** [review]
Great work! LGTM

**Pull request merged**